raigad/src/main/java/com/netflix/raigad/monitoring/HealthMonitor.java

```diff
@@ -61,13 +65,19 @@ public HealthMonitor(IConfiguration config,InstanceManager instanceManager)
     @Override
     public void execute() throws Exception {
 
-        // If Elasticsearch is started then only start the monitoring
-        if (!ElasticsearchProcessMonitor.isElasticsearchStarted()) {
-            String exceptionMsg = "Elasticsearch is not yet started, check back again later";
+        // report cluster-metrics only if Elasticsearch process is running
+        if (!ElasticsearchProcessMonitor.isElasticsearchRunning()) {
+            String exceptionMsg = "Elasticsearch is not running, check back again later";
             logger.info(exceptionMsg);
             return;
         }
 
+        // In case we configured only the Master-node to report metrics
+        // and this node is not a master - bail out
+        if (config.reportMetricsFromMasterOnly() && EsUtils.amIMasterNode(config, httpModule)) {
+            return;
```

Comment:
This check will work for all nodes except Master. 
I think, it should be as follows
if (config.reportMetricsFromMasterOnly() && !EsUtils.amIMasterNode(config, httpModule))